frontend/app/src/App.tsx

```diff
@@ -1683,7 +1685,15 @@ export class App extends PureComponent<Props, State> {
             attach={window}
             focused={true}
           >
-            <StyledApp className={outerDivClass}>
+            <StyledApp
+              className={outerDivClass}
+              data-testid="stApp"
+              data-teststate={
+                scriptRunId == INITIAL_SCRIPT_RUN_ID
+                  ? "initial"
+                  : scriptRunState
+              }
+            >
```

Comment:
Need a review of this change.

I added data-teststate to make our tests less brittle. So instead of checking whether the status text in the top right of the page says "Running" we can see if `data-teststate` is in "Running" state.

The big win here is that we can now well "hasn't even ran once" (i.e. the "initial" state) from "has ran once and is done now" (i.e. the "notRunning" state).

Reply:
Looks good 👍  We also have the same kind of check for playwright e2e tests here: 

https://github.com/streamlit/streamlit/blob/3118dbdee3c30c8a259868014899fa36e5da71c9/e2e_playwright/conftest.py#L180-L185

But I can also update this to use `data-teststate` once this is merged.